client/src/app/site/mediafiles/components/mediafile-list/mediafile-list.component.ts

```diff
@@ -415,22 +445,50 @@ export class MediafileListComponent extends BaseViewComponent implements OnInit,
                     parent_id: this.directory ? this.directory.id : null,
                     is_directory: true
                 });
-                this.repo.create(mediafile).then(null, this.raiseError);
+                this.repo.create(mediafile).catch(this.raiseError);
             }
         });
     }
 
-    public move(templateRef: TemplateRef<string>, mediafile: ViewMediafile): void {
-        this.newDirectoryForm.reset();
+    public move(templateRef: TemplateRef<string>, mediafiles: ViewMediafile[]): void {
+        this.moveForm.reset();
+
+        if (mediafiles.some(file => file.is_directory)) {
+            this.filteredDirectoryBehaviorSubject.next(
+                this.directoryBehaviorSubject.value.filter(
+                    dir => !mediafiles.some(file => dir.path.startsWith(file.path))
```

Comment:
@FinnStutzenstein I found a pretty simple solution to avoid using trees, since the path is already present and should be unique enough